packages/snap-store-mobx/src/Search/Stores/SearchResultStore.ts

```diff
@@ -181,17 +181,17 @@ export class Variants {
 	public selections: VariantSelection[] = [];
 	public setActive: (variant: Variant) => void;
 
-	constructor(variantData: VariantData[], mask: ProductMask, preselectedOptions?: Record<string, string[]>) {
+	constructor(variantData: VariantData[], mask: ProductMask, config?: VariantConfig) {
 		// setting function in constructor to prevent exposing mask as class property
 		this.setActive = (variant: Variant) => {
 			this.active = variant;
 			mask.set({ mappings: this.active.mappings, attributes: this.active.attributes });
 		};
 
-		this.update(variantData, preselectedOptions);
+		this.update(variantData, config);
 	}
 
-	public update(variantData: VariantData[], preselectedOptions?: Record<string, string[]>) {
+	public update(variantData: VariantData[], config?: VariantConfig) {
```

Comment:
If this function is called externally (in middleware for example), will the user need to pass the `config` parameter used at construction to get the same settings?